metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java

```diff
@@ -100,7 +107,16 @@ public SearchResponse search(SearchRequest searchRequest) throws InvalidSearchEx
     searchResponse.setResults(Arrays.stream(elasticsearchResponse.getHits().getHits()).map(searchHit -> {
       SearchResult searchResult = new SearchResult();
       searchResult.setId(searchHit.getId());
-      searchResult.setSource(searchHit.getSource());
+      Map<String, Object> source;
+      if (fields.isPresent()) {
+        source = new HashMap<>();
+        searchHit.getFields().forEach((key, value) -> {
+          source.put(key, value.getValues().size() == 1 ? value.getValue() : value.getValues());
```

Comment:
@merrimanr Do you have an integration test that checks when the result size is 0? I didn't notice one. I'm not familiar enough with `value.getValues().size()` to know how this works for boundary conditions. The ES javadoc doesn't really provide details.